src/vs/workbench/contrib/terminal/browser/media/shellIntegration-fish.fish

```diff
@@ -0,0 +1,91 @@
+# ----------------------------------------------------------------------------
+#   Visual Studio Code terminal integration for fish
+#   Copyright (c) Microsoft Corporation. All rights reserved.
+#   Licensed under the MIT License. See License.txt in the project root for license information.
+# ----------------------------------------------------------------------------
+# Manual installation:
+#
+#   (1) Add the following to the end of `$__fish_config_dir/config.fish`:
+#
+#         string match -q "$TERM_PROGRAM" "vscode"
+#         and . (code --locate-shell-integration-path fish)
+#
+#   (2) Restart fish.
+# ----------------------------------------------------------------------------
+# TODO: Confirm all escape sequences once they are finalized.
+# See microsoft/vscode#155639 and microsoft/vscode#139400 for discussion.
+# ----------------------------------------------------------------------------
+
+# Don't run in scripts, other terminals, or more than once per session.
+status is-interactive
+and string match --quiet "$TERM_PROGRAM" "vscode"
+and ! set --query VSCODE_SHELL_INTEGRATION
+or return
+
+set --global VSCODE_SHELL_INTEGRATION 1
+
+# Helper function
+function __vsc_esc -d "Emit escape sequences for VS Code shell integration"
+	builtin printf "\e]633;%s\007" (string join ";" $argv)
+end
+
+# Sent right before executing an interactive command.
+# Marks the beginning of command output.
+function __vsc_cmd_executed --on-event fish_preexec
+	__vsc_esc C
+	__vsc_esc E "$argv"
+end
+
+# Sent right after an interactive command has finished executing.
+# Marks the end of command output.
+function __vsc_cmd_finished --on-event fish_postexec
+	__vsc_esc D $status
+end
```

Comment:
If no command ran, just `__vsc_esc D` should run without an exit code. That will make the below be circle outlines instead of blue:

<img width="307" alt="Screen Shot 2022-08-08 at 3 53 58 pm" src="https://user-images.githubusercontent.com/2193314/183528070-d766afcb-a68e-4f3e-9c73-5f08c4f9d769.png">

Reply:
This appears to have been addressed by adding `__vsc_cmd_clear` hooked into the `fish_cancel` event.

Reply:
@zgracem This doesn't seem to be completely resolved. This is zsh:

<img width="234" alt="image" src="https://user-images.githubusercontent.com/44045911/183869316-3e8391eb-458d-4610-87fa-1be3e919e534.png">

And this is with Fish script from this PR:

<img width="643" alt="image" src="https://user-images.githubusercontent.com/44045911/183868111-01cc4723-0d88-4068-adfa-dc794e39be26.png">

As you can see the `fish_cancel` event is only fired for Ctrl+C but not when the commandline is empty, which only triggers a `fish_prompt` event. My plugin added [a extra check](https://github.com/kidonng/vscode.fish/commit/31745a39f90f384c6f8c184a03c1ac9e43ae55ba) for this.

Reply:
@kidonng nice observation, do you want to send a PR since it's your solution? I can fix it up otherwise

Reply:
I'm glad to!